spec/lang/well-formed.md

```diff
@@ -759,6 +776,16 @@ impl Program {
             }
         }
 
+        // Check vtables: All vtables for the same trait must have the same methods defined.
+        let mut traits: Map<TraitName, Set<TraitMethodName>> = Map::new();
+        for (_name, vtable) in self.vtables {
+            let methods = vtable.methods.keys().collect();
+            match traits.insert(vtable.trait_name, methods) {
+                Some(prev_methods) => ensure_wf(methods == prev_methods, "Program: vtables with same trait but different methods")?,
```

Comment:
Hm, should the program contain a `traits: Map<TraitName, Set<TraitMethodName>>` field so that this becomes a bit less ad-hoc here?

Reply:
I think a comment can easily explain that confusion.

Having some central place where all (dynamically invokeable) traits are "declared" makes a lot of sense IMO.